## Capítulo 1.1.: Modelos estadísticos para control de calidad

### Caso 1.1.1: Análisis de capacidad de proceso

In [7]:
import numpy as np
import math
import scipy.stats as st

### Ejemplo 3:

Una característica de calidad importante en la fabricación de una llanta es la longitud de capa, que para cierto tipo de llanta debe ser de 870 mm con una tolerancia de $\pm$ 7 mm. Para monitorear el correcto funcionamiento del proceso de corte, cada media hora se toman cinco (5) capas y se miden. De acuerdo a las mediciones realizadas el último mes, en donde el proceso ha estado trabajando de manera estable, se tiene que la media y la desviación estándar del proceso son $\mu =$ 873 mm y $\sigma =$ 2.5 mm, respectivamente. Con base a lo anterior se quiere saber en qué medida el proceso está cumpliendo con las especificaciones respondiendo a los siguientes interrogantes: 
* **a) Determine la capacidad del proceso (Cp).**
* **b) ¿Cuál es la capacidad real del proceso (Cpk)?**
* **c) Calcule los límites de fluctiación natural del proceso (LFNP).**
* **d) ¿Qué proporción de producto no conforme se está generando por fuera de los límites de especificación?**
* **e) ¿Entre qué valores debería oscilar la media del proceso para que el porcentaje de no conformidad sea a lo sumo 4.0%?**

**a) Determine la capacidad del proceso (Cp).**

R/ Se sabe que el índice de capacidad del proceso Cp esta dado por:

$$\rightarrow C_{p}=\frac{T}{6\sigma'}$$

Y:
$$\rightarrow T = 2\nabla = LSE - LIE$$

Donde:
* $\nabla$ es la discrepancia y de ahora en adelante se le asignará la letra 'D', 
* $\sigma'$ que es la desviación estándar se le asignará la letra 's',y 
* 'T' es la tolerancia. 

A continuación se procederá determinar la capacidad del proceso:

In [5]:
LIE = 870 - 7 # Límite Inferior de Especificación
LSE = 870 + 7 # Límite Superior de Especificación

T = LSE - LIE

s = 2.5

Cp = T/(6*s)
print(Cp)

0.9333333333333333


Dado que $0.67 < C_{p} < 1.00$ entonces esto por lo pronto indica que el proceso no tiene capacidad de fabricar producto dentro de las especificaciones, requiere modificaciones para generar producto que alcance una calidad satisfactoria.

**b) ¿Cuál es la capacidad real del proceso (Cpk)?**

R/ Cuando el proceso no está centrado, se emplea el **índice de capacidad unilateral** o **índice de capacidad real (Cpk)**:

$$\rightarrow C_{pk}=min(C_{pki}, C_{pks})$$

$$\rightarrow C_{pk}=min(C_{pki}=\frac{\bar{X'}-LIE}{3\sigma'}, C_{pks}=\frac{LSE-\bar{X'}}{3\sigma'})$$

Donde:

* $\bar{X'}$ es la media del proceso, y
* $\sigma'$ es la desviación del proceso.

Por tanto, el índice Cpk es:

In [8]:
X = 873

Cpki = np.round((X-LIE)/(3*s),2)
Cpks = np.round((LSE-X)/(3*s),2)

Cpk = np.min([Cpki,Cpks])

print("Cpki = "+ str(Cpki))
print("Cpks = "+ str(Cpks))
print("Cpk = "+ str(Cpk))

Cpki = 1.33
Cpks = 0.53
Cpk = 0.53


Dado que Cpk $\approx$ 0.53, esto indica que el proceso no es realmente capaz y por lo menos tiende a no cumplir con una de las especificaciones de calidad. Dicho lo anterior, tal parece que no se está cumpliendo con el límite superior de calidad, pues corresponde a Cpks $\approx$ 0.53. Por el contrario, el proceso no parece tener problemas para cumplir con el límite inferior de especificación, ya que Cpki $\approx$ 1.33.

**c) Calcule los límites de fluctuación natural del proceso (LFNP):**

R/ Se tiene que los límites de fluctuación natural del proceso están dados por la siguiente fórmula:

$$\rightarrow LFN_{i,s} = \bar{X'} \pm 3\sigma'$$

Donde:

* $\bar{X'}$ es la media del proceso, y
* $\sigma'$ es la desviación estándar del proceso.

Entonces, los límites de fluctuación son:

In [9]:
LIFN = np.round(X - 3*s,4)
LSFN = np.round(X + 3*s,4)

print("LIFN = " + str(LIFN) + " y LSFN = " + str(LSFN))

LIFN = 865.5 y LSFN = 880.5


Se compara si LIFN es mayor o igual a LIE y si LSFN es menor o igual a LSE:

In [10]:
print(LIFN>=LIE, LSFN<=LSE)

True False


Se tiene que **el límite superior fluctuación natural (LIFN) es mas alto (>) que el límite superior de especificación (LSE)**, por lo que **se confirma que el proceso está generando producto no conforme por encima de dicho límite.**

**d) ¿Qué proporción de producto no conforme se está generando por fuera de los límites de especificación?**

R/ Como el proceso está generando producto no conforme por encima del límite superior de especificación, inicialmente se debe determinar el valor Z estandarizado para dicho límite. Esto se obtiene con la siguiente fórmula:

$$\rightarrow Z_{sup}=\frac{LSE-\bar{X'}}{\sigma'}$$

Obteniendo el siguiente resultado:

In [23]:
Zsup = np.round((LSE - X)/s,4)
print(Zinf)

1.6


Luego de esto, se obtiene de la tabla de distribución normal estándar la probabilidad de obtener un valor Zsup mayor a 1.6, así:

In [19]:
P = np.round(1 - st.norm.cdf(Zinf),4)
print(P)

0.0548


Es decir que la proporción de producto no conforme generado por encima del límite superior de especificación es del 5.48%. El porcentaje es bajo pero no despreciable, aproximadamente 548 unidades defectuosas en cada lote de 10.000 unidades, pero se podría mejorar centrando un poco más el proceso.

**e) ¿Entre qué valores debería oscilar la media del proceso para que el porcentaje de no conformidad sea a lo sumo 4.0%?**

Inicialmente se debe despejar $\bar{X'}$ de la fórmula de Z estandarizado inferior, así:

$$Z_{inf}=\frac{LIE-\bar{X'}}{\sigma'}\rightarrow\bar{X'}_{inf}=LIE-\sigma'Z_{inf}$$

Lo mismo ocurre con la formula de Z estandarizado superior:

$$Z_{sup}=\frac{LSE-\bar{X'}}{\sigma'}\rightarrow\bar{X'}_{sup}=LSE-\sigma'Z_{sup}$$

Donde:
* $Z_{inf}$ es el valor de Z estandarizado inferior que representa una probabilidad de 4.0% de obtener producto no conforme por debajo de LIE, y
* $Z_{sup}$ es el valor de Z estandarizado superior que representa una probabilidad de 96.0% (100% - 4%) de no obtener inconformidades de producto por encima de LSE.

Por lo tanto:

In [20]:
VO_inf = 0.04
Zinf_vo = np.round(st.norm.ppf(VO_inf),4)


VO_sup = 1 - VO_inf
Zsup_vo = np.round(st.norm.ppf(VO_sup),4)

print("Zinf = "+str(Zinf_vo) + " y Zsup = " + str(Zsup_vo))

Zinf = -1.7507 y Zsup = 1.7507


Entonces, los valores entre los cuales puede oscilar $\bar{X'}$ para obtener a lo sumo 4.0% de producto no conforme son:

In [22]:
Xinf_vo = np.round(LIE - (Zinf_vo*s),1) # Cuando el proceso entrega 4% de producto no conforme por debajo del límite inferior de especificación.
Xsup_vo = np.round(LSE - (Zsup_vo*s),1) # Cuando el proceso entrega 4% de producto no conforme por encima del límite superior de especificación.
print("Xinf = " + str(Xinf_vo) + " y Xsup = " + str(Xsup_vo))

Xinf = 867.4 y Xsup = 872.6
